allure-behave/src/listener.py

```diff
@@ -97,8 +101,11 @@ def stop_test(self, parent_uuid, uuid, name, context, exc_type, exc_val, exc_tb)
         self.stop_scenario(context['scenario'])
 
     def stop_scenario(self, scenario):
-        should_run = (scenario.should_run_with_tags(self.behave_config.tags) and
-                      scenario.should_run_with_name_select(self.behave_config))
+        if BEHAVE_1_2_7_OR_GREATER:
+            should_run = scenario.should_run_with_tags(self.behave_config.tag_expression)
+        else:
+            should_run = scenario.should_run_with_tags(self.behave_config.tags)
+        should_run &= scenario.should_run_with_name_select(self.behave_config)
```

Comment:
Let's use a boolean operator instead of a bitwise one. It's not uncommon in Python to return true-ish values (e.g., non-zero numbers or non-empty strings) instead of True. & might yield invalid results in such cases:

```python
assert 1 and 2
assert not(1 & 2)
```

I agree that currently, `should_run_with_name_select` returns a boolean value, so there is no difference between `&` and `and`. But IMO, it's better not to rely on that (at least until Behave includes the return type to the contract via typings).

```suggestion
        should_run = should_run and scenario.should_run_with_name_select(self.behave_config)
```

Reply:
Sounds good to me.